src/rules/valid-bin.ts

```diff
@@ -17,17 +22,45 @@ export const rule = createRule({
 				);
 
 				const errors = validateBin(binValue);
-				if (!errors.length) {
-					return;
+				if (errors.length) {
+					context.report({
+						data: {
+							errors: formatErrors(errors),
+						},
+						messageId: "validationError",
+						node: binValueNode,
+					});
 				}
 
-				context.report({
-					data: {
-						errors: formatErrors(errors),
-					},
-					messageId: "validationError",
-					node: binValueNode,
-				});
+				if (
+					shouldEnforceCase &&
+					node.value.type === "JSONObjectExpression"
+				) {
+					for (const property of node.value.properties) {
+						const key = property.key as JsonAST.JSONStringLiteral;
```

Comment:
👍 yeah the string keys will never be numeric literals or identifiers. I wonder if jsonc-eslint-parser's types could be augmented to indicate the AST is only valid JSON, not `.jsonc`/`.json5`?

Not a priority at all IMO.